odincal/odincal/odincal/level1b_window_importer2.py

```
@@ -0,0 +1,385 @@
+        return
+
+    processtemp = {'file': acfile,
+                   'info': info['info'],
+                   'total_scans': info['total_scans'],
+                   'success_scans': info['success_scans'],
+                   'version': info['version']}
+    con.query('''delete from processed
+                     where file='{0}' and version={1} '''.format(*temp))
+    con.insert('processed', processtemp)
+
+
+def preprocess_data(acfile, backend, version, con, stw1, stw2, tdiff, logger):
+    """prepara data for calibration, i.e import data
+       to database tables"""
+
+    prepare_data = PrepareData(acfile, backend, version, con)
+
+    # find out which sodaversion we have for this data
+    sodaversion = prepare_data.get_soda_version(stw1, stw2)
+    if sodaversion == -1:
+        info = {'info': 'no attitude data',
+                'total_scans': 0,
+                'success_scans': 0,
+                'version': version}
+        report_result(con, acfile, info)
+        logger.warning(
+            'no imported level0 attitude data found for processing file {0}'.format(acfile))  # noqa
+        return 1
+
+    # perform level0 data processing
+    error = prepare_data.att_level1_process(
+        stw1 - tdiff, stw2 + tdiff, sodaversion)
+    if error == 1:
+        report_result(con, acfile, {'info': 'pg problem'})
+        return 1
+
+    error = prepare_data.shk_level1_process(
+        stw1 - tdiff, stw2 + tdiff)
+    if error == 1:
+        report_result(con, acfile, {'info': 'pg problem'})
+        return 1
+
+    error = prepare_data.ac_level1a_process(
+        stw1 - tdiff, stw2 + tdiff)
+    if error == 1:
+        report_result(con, acfile, {'info': 'pg problem'})
+        return 1
+
+    return 0
+
+
+class Level1BImportError(Exception):
+    pass
+
+
+class Level1BPrepareDataError(Level1BImportError):
+    pass
+
+
+class Level1BPreprocessDataError(Level1BImportError):
+    pass
+
+
+def level1b_importer(acfile, backend, version, con):
```

This function now raises instead of returning 1 on error, and takes a connection as an argument.